### PR TITLE
Remove the old datasplitters from EvalML

### LINKED ISSUE #2099: Remove old datasplitters
with #2098 and #2014, we add oversampling/undersampling support to AutoML. Since we decide to do sampling after feature generation, our previous [PR that added sampling data splitters](https://github.com/alteryx/evalml/commit/80267d7ac1f5f7ffc8bb16a27309c8fb105e5d6e) is no longer needed. We can delete these old samplers.

### PR DESCRIPTION
**Pull request by bchen1116**
Fix #2099 

Note: I will likely merge this around the same time I merge [2128](https://github.com/alteryx/evalml/pull/2128), which adds the sampler components to AutoMLSearch. This way, we will still maintain support for imbalanced datasets

### CODE REVIEW

File: docs/source/release_notes.rst [hunk 1]
```diff
@@ -8,6 +8,7 @@ Release Notes
     * Fixes
     * Changes
             * Updated pipeline API to accept component graph and other class attributes as instance parameters. Old pipeline API still works but will not be supported long-term. :pr:`2091`
+            * Removed all old datasplitters from EvalML :pr:`2193`
```
Comment (jeremyliweishih) [inline docs/source/release_notes.rst:11]: Does this count as a breaking change? Might be a good idea to add.

**Review by jeremyliweishih — APPROVED** [review]
This looks good from my end. Must have been satisfying to delete all these files

**Review by chukarsten — APPROVED** [review]
RIP

**Review by angela97lin — APPROVED** [review]
✂️ ✂️ 👍

**Review by dsherry — APPROVED** [review]
Looks good to me!

**Comment by dsherry** [comment]
@bchen1116 oh I think you should also delete `evalml/preprocessing/data_splitters/base_splitters.py`.

**Comment by dsherry** [comment]
For those following along: the plan is to merge #2128 first, then this PR, and then that unblocks the oversampler work #2213

**Pull request merged**